lib/internal/Magento/Framework/Image/Adapter/Gd2.php

```diff
@@ -465,7 +465,7 @@ public function watermark($imagePath, $positionX = 0, $positionY = 0, $opacity =
         } elseif ($this->getWatermarkPosition() == self::POSITION_CENTER) {
             $positionX = $this->_imageSrcWidth / 2 - imagesx($watermark) / 2;
             $positionY = $this->_imageSrcHeight / 2 - imagesy($watermark) / 2;
-            imagecopymerge(
+            $this->imagecopymergeWithAlphaFix(
```

Comment:
I believe the proper name for this function should be something like `copyImage` as we are not even using `imagecopymerge` anymore.